Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1391,14 +1405,35 @@ void MediaPlayerPrivateGStreamerBase::dispatchDecryptionKey(GstBuffer* buffer)
     GST_TRACE("emitted decryption cipher key on pipeline, event handled %s, need to resend credentials %s", boolForPrinting(eventHandled), boolForPrinting(m_needToResendCredentials));
 }
 
-void MediaPlayerPrivateGStreamerBase::dispatchDecryptionSession(const String& sessionId)
+#if USE(OPENCDM)
+bool MediaPlayerPrivateGStreamerBase::findAndSetPendingProtectionEventByInitData(const Vector<uint8_t>& initData, uint32_t protectionEvent)
 {
-    bool eventHandled = gst_element_send_event(m_pipeline.get(), gst_event_new_custom(GST_EVENT_CUSTOM_DOWNSTREAM_OOB,
-        gst_structure_new("drm-session", "session", G_TYPE_STRING, sessionId.utf8().data(), nullptr)));
-    m_needToResendCredentials = m_handledProtectionEvents.size() > 0;
-    GST_TRACE("emitted decryption session %s on pipeline, event handled %s, need to resend credentials %s", sessionId.utf8().data(), boolForPrinting(eventHandled), boolForPrinting(m_needToResendCredentials));
+    for (auto& initDataProtectionEventsMapping : m_initDataProtectionEventsMapping) {
+        if (initDataProtectionEventsMapping.first == initData) {
+            initDataProtectionEventsMapping.second.add(protectionEvent);
+            return true;
+        }
+    }
+
+    return false;
 }
 
+void MediaPlayerPrivateGStreamerBase::dispatchOrStoreDecryptionSession(const String& sessionId, const unsigned& protectionEvent)
+{
+    if (m_triggeredProtectionEvents.contains(protectionEvent)) {
```

Comment:
You didn't honor the comment I made about the flow and how we should consider the events handled. Please read it and change this code to honor it or discuss why this is wrong.

Reply:
We are really sorry for making a misunderstanding here. We have honored all comments and updated accordingly. While addressing some of your comments we have seen the logic problems, so that we handled based on our understanding and added reply against your comments. Actually we would like to discuss with you these comments before doing the commits, but unfortunately you were on vacation, so we were not able to discuss with you and that is why only I have added reply against these type of comments. I think we can discuss it in opencdm flow and we will update it.